encoders/src/main/scala/au/csiro/pathling/encoders/SerializerBuilder.scala

```diff
@@ -226,16 +224,22 @@ private[encoders] object SerializerBuilderProcessor {
     // Primitive single-value types typically use the Element suffix in their
     // accessors, with the exception of the "div" field for reasons that are not clear.
     //noinspection DuplicatedCode
-    if (field.isInstanceOf[RuntimeChildPrimitiveDatatypeDefinition] &&
-      field.getMax == 1 &&
-      field.getElementName != "div")
-      "get" + field.getElementName.capitalize + "Element"
-    else {
-      if (field.getElementName.equals("class")) {
-        "get" + field.getElementName.capitalize + "_"
-      } else {
+    field match {
+      case p: RuntimeChildPrimitiveDatatypeDefinition if p.getMax == 1 && p
+        .getElementName != "div" =>
+        if ("reference" == p.getElementName && classOf[IBaseReference]
+          .isAssignableFrom(p.getField.getDeclaringClass)) {
+          // special case for subclasses of IBaseReference
+          // for some obscure reason the accessor getReferenceElement returns IdType rather than StringType and 
+          // getReferenceElement_ needs to be used instead.
+          "getReferenceElement_"
```

Comment:
If we are using `getReferenceElement_`, should we be checking if it is assignable from `Reference`, not `IBaseReference`?

Reply:
Possibly. 
`Reference` is a FHIR version specific class, so if we were to use `r4.Reference` we technically (to keep the basic SerializerBuilder version independent) should delegate this to R4TypeMappings.
Using 'IBaseReference' is a compromise as all the Reference types inherit from IBaseReference and starting from 'stu3'  have the `getReferenceElement_()` method.

Reply:
Ok, thanks.

I think this solution is fine as long as we have test coverage that verifies that it works.